components/skeleton/test/skeleton-group.vdiff.js

```diff
@@ -0,0 +1,235 @@
+`;
+
+const noSkeletonFixture = html`
+	<d2l-skeleton-group-test-wrapper style="width: 330px">
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+		<d2l-test-skeleton-container></d2l-test-skeleton-container>
+	</d2l-skeleton-group-test-wrapper>
+`;
+
+const nestedGroupsRemoveAllFixture = html`
+	<div style="width: 330px">
+		<d2l-skeleton-group-test-wrapper id="to-remove" skeleton>
+			<d2l-skeleton-group-test-wrapper id="to-remove" skeleton>
+				<d2l-test-skeleton-heading level="1">Heading 1</d2l-test-skeleton-heading>
+				<d2l-test-skeleton-box id="to-remove" skeleton></d2l-test-skeleton-box>
+				<d2l-test-skeleton-container></d2l-test-skeleton-container>
+			</d2l-skeleton-group-test-wrapper>
+			<d2l-skeleton-group-test-wrapper>
+				<d2l-test-skeleton-box></d2l-test-skeleton-box>
+			</d2l-skeleton-group-test-wrapper>
+		</d2l-skeleton-group-test-wrapper>
+	</div>
+`;
+
+describe('d2l-skeleton-group', () => {
+
+	it('all-skeleton', async() => {
+		const elem = await fixture(html`
+			<d2l-skeleton-group-test-wrapper style="width: 330px">
+				<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+				<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+				<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>
+			</d2l-skeleton-group-test-wrapper>
+		`);
+		await expect(elem).to.be.golden();
+	});
+
+	it('one-skeleton', async() => {
+		const elem = await fixture(oneSkeletonFixture);
+		await expect(elem).to.be.golden();
+	});
+
+	it('no-skeleton', async() => {
+		const elem = await fixture(noSkeletonFixture);
+		await expect(elem).to.be.golden();
+	});
+
+	it('make-one-skeleton', async() => {
+		const elem = await fixture(noSkeletonFixture);
+		const container = elem.querySelector('d2l-test-skeleton-container');
+		container.skeleton = true;
+		await container.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	it('make-one-not-skeleton', async() => {
+		const elem = await fixture(oneSkeletonFixture);
+		const container = elem.querySelector('d2l-test-skeleton-container');
+		container.skeleton = false;
+		await container.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	it('add-element', async() => {
+		const elem = await fixture(noSkeletonFixture);
+		elem.insertAdjacentHTML('beforeend', '<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>');
+		await elem.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	it('remove-element', async() => {
+		const elem = await fixture(oneSkeletonFixture);
+		elem.querySelector('d2l-test-skeleton-container[skeleton]').remove();
+		await elem.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	[true, false].forEach((skeleton) => {
+		it(`mixed-elements${skeleton ? '-skeleton' : ''}`, async() => {
+			const elem = await fixture(html`
+				<d2l-skeleton-group-test-wrapper style="width: 330px">
```

Comment:
is hard coding `width` the new best practice?

Reply:
It was what I needed to do here to get clean diffs. An alternate would be to set the viewport, but when I did that the responsive rules for typography (headings specifically) kicked in so all the headings were smaller.

Probably the "best" best practice would be to just not set a width or viewport and just roll with the default 800px width, unless that's totally unreasonable.

Reply:
yeah. 😬 I think I did this because of some weirdness with the `*-test-wrapper` functions that were baked in, and iirc, it wasn't trivial to change so I moved on (and forgot about it 😆).

I might take a look at improving that at some point